Automattic/jetpack
# WordAds: Enable A8C CMP banner support

## Proposed changes:
Adds the support for loading the A8C TCF 2.2 compatible CMP banner on Jetpack sites enrolled in WordAds. This allows collecting consent for personalized ads from site visitors under GDPR laws.

### Other information:


## Jetpack product discussion
N/A



## Does this pull request change what data or activity we track or use?
N/A



## Testing instructions:

* Enroll your test Jetpack site to WordAds, if you have not enrolled already.
* Comment out this `wordads_cmp_enabled` check https://github.com/Automattic/jetpack/commit/1987516e4df5f00956ad8bff71a77a63f604df59 (as this option is available in a different branch https://github.com/Automattic/jetpack/pull/35205)
* Visit your test Jetpack site from an EU country (you might need a VPN for this)
* The CMP banner should appear and you should be able to manage your privacy preference.
* Confirm that ad requests respect privacy preference.
<img width="1728" alt="cmp-on-jetpack" src="https://github.com/Automattic/jetpack/assets/9039613/c616b5e5-e388-4c93-b8ed-b24c734aa249">
<img width="1723" alt="cmp-on-jetpack-2" src="https://github.com/Automattic/jetpack/assets/9039613/547afe76-d069-4e1f-9f15-99c5f11511d3">



**Confirm no conflict with Cookie consent block**
- First, enable the [Extra Sidebar Widgets](https://jetpack.com/support/extra-sidebar-widgets/), so the block can be available on your site.
- Add the cookie consent block to your site using [this guide](https://wordpress.com/support/wordpress-editor/blocks/cookie-consent-block/).
- Visit your site and confirm the Cookie consent block is NOT visible and is overridden by the CMP GDPR banner
- Now deactivate the GDPR banner from running e.g by NOT commenting out this `wordads_cmp_enabled` check this time https://github.com/Automattic/jetpack/commit/1987516e4df5f00956ad8bff71a77a63f604df59 (as this option is available in a different branch https://github.com/Automattic/jetpack/pull/35205)
- Visit your site again and this time the Cookie consent block should be visible
<img width="1728" alt="cookie consent" src="https://github.com/Automattic/jetpack/assets/9039613/a4149ea7-0d06-4285-a349-75accfabdf89">
